.github/workflows/test.yml

```diff
@@ -91,12 +93,6 @@ jobs:
           # suggesting no NumPy when there is
           pip install BoltzTraP2
 
-      - name: Install pymatgen
-        run: |
-          micromamba activate pmg
-
-          uv pip install --editable '.[${{ matrix.config.extras }}]' --resolution=${{ matrix.config.resolution }}
```

Comment:
Current we cannot put this before `pip install BoltzTraP2` (`BoltzTraP2` is included as an optional dependency) because `uv` cannot install `BoltzTraP2` for some reason.

Meanwhile I'm not quite sure why `pip install BoltzTraP2` failed in Windows either, I would look closer into this and fix it in a follow up PR if possible :)

Meanwhile I think we should not include 57d34e2e297a0669534692cc9154dde1bc66dadf here as a solution has been proposed in #3972 and we shouldn't just hijack his contribution :)

Reply:
ah, hadn't seen https://github.com/materialsproject/pymatgen/pull/3972 yet.


can you move `pip install BoltzTraP2` into the preceding step then and add an inline comment explaining order matters?

also, let's use `matplotlib!=3.9.1` instead of `matplotlib<3.9.1` in pyproject

Reply:
Sure thing.

Reply:
But things are getting quite tricky at this moment (surrounding `BoltzTraP2`), as it would require `cython` to be built properly (I suggested to the original author to distribute wheels but wasn't making any progress so far). Perhaps we can comment out `BoltzTraP2` from optional dependency for now?

Reply:
i think that's fine? we can also move `uv pip install cython setuptools wheel` further up

Reply:
Beautiful. That's better :)

Reply:
Meanwhile want to know if it's possible to insert in-line OS condition (Ubuntu only)? I googled earlier and didn't have much luck. `BoltzTraP2` is only installable on Ubuntu at this moment.

Reply:
> I now cannot think of another way to install those dependencies other than the one prior to https://github.com/materialsproject/pymatgen/commit/155cbff5e01ab2108a9bf831ec6f73004b85fb8a. Perhaps you could push that directly? :)

maybe it's not possible then. sorry for messing up your PR. i'll take another look later

> Meanwhile want to know if it's possible to insert in-line OS condition (Ubuntu only) in test.yml? I googled earlier and didn't have much luck.

have a look at this [SO](https://stackoverflow.com/a/8597411)

Reply:
I just [pushed a MR (merge request in GitLab) to BoltzTraP2](https://gitlab.com/sousaw/BoltzTraP2/-/merge_requests/21) which should fix the installation issue with `uv`, and the original author is working on [fixing the installation on Windows as well](https://gitlab.com/sousaw/BoltzTraP2/-/issues/5). This would take some time but at least there is hope :)

Reply:
nice work! that's what i call community effort 👍 😄